packages/core/src/browser/label-provider.ts

```diff
@@ -98,14 +99,20 @@ export class DefaultUriLabelProviderContribution implements LabelProviderContrib
 }
 
 @injectable()
-export class LabelProvider {
+export class LabelProvider implements FrontendApplicationContribution {
```

Comment:
I could extract another class to avoid adding `initialize` as public API to `LabelProvider`, something like `LabelProviderContributions`. `LabelProvider` will use it to `find` contributions listen to its changes.